pkg/controller/job/job_controller.go

```diff
@@ -660,9 +695,12 @@ func pastBackoffLimitOnFailure(job *batch.Job, pods []*v1.Pod) bool {
 	return result >= *job.Spec.BackoffLimit
 }
 
-// pastActiveDeadline checks if job has ActiveDeadlineSeconds field set and if it is exceeded.
+// pastActiveDeadline checks if job has ActiveDeadlineSeconds field set and if
+// it is exceeded. If the job is currently suspended, the function will always
+// return false.
 func pastActiveDeadline(job *batch.Job) bool {
-	if job.Spec.ActiveDeadlineSeconds == nil || job.Status.StartTime == nil {
+	suspended := utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) && job.Spec.Suspend != nil && *job.Spec.Suspend
```

Comment:
create a `jobSuspended` function and use it directly in the if statement below, ditto for other places.

Reply:
Thanks, done.